app/components/generic-chart.js

```diff
@@ -151,139 +21,132 @@ export default Ember.Component.extend({
         this.updateChart();
     }),
 
-    updateChart() {
-        switch(this.get('chartType')) {
-            case 'donut-chart':{
-                this.set('data', this.get('aggregations.sources.buckets'));
-                let columns = this.get('sourcesList');
-                let title = 'Published in...';
-                this.$(`#timeseriesCheckbox`).hide();
-                this.drawDonut(title, columns);
-                break;}
-            case 'bar-chart':{
-                this.set('data', this.get('aggregations.contributors.buckets'));
-                let columns = this.get('sourcesList'); // jscs:ignore
-                let title = 'Top 10 Contributors: ';
-                this.$(`#timeseriesCheckbox`).hide();
-                this.drawBar(title, columns);
-                break;}
-            case 'timeseries-chart':{
-                this.set('data', this.get('aggregations.articles_over_time.buckets'));
-                let columns = this.get('timeseriesList');
-                let title = '';
-                let interval = this.get('interval');
-                this.$(`#timeseriesCheckbox`).show();
-                this.drawTS(title, columns, interval);
-                break;}
-        }
-    },
+    operationDict: {'donut-chart':{'title':'Published in...',
+                                  'columns':function(data) {
+                                      return data.map(({ key, doc_count }) => [key, doc_count]);},
+                                  'c3_plot':function(title, columns, element, height, width){
+                                      let donut = c3.generate({
+                                          bindto: element,
+                                          data: {
+                                              columns,
+                                              type: 'donut'
+                                          },
+                                          legend: { show: false },
+                                          donut: {
+                                              title,
+                                              label: {
+                                                  show: false
+                                              }
+                                          },
+                                          size: { height: height*150-20,
+                                                  width: width*150
+                                                },
+                                      });
+                                  },
+                                },
+                    'bar-chart':{'title':'Top 10 Contributors',
+                                 'columns':function(data) {
+                                       return data.map(({ key, doc_count }) => [key, doc_count]).slice(0, 10);},
```

Comment:
Here, if we only want the top ten, we should limit it to the top ten in the aggregation.